### DISCUSSION TITLE
Adding maps without internet

### DISCUSSION THREAD

**Discussion by vb0** [osmandapp/OsmAnd#13841] [Q&A] [answered]
What's the way to refresh local maps if adding some obf's by hand? I mean I just can't figure out how to re-read the local storage and tell OsmAnd they're there. I know I need to take out the _2 from the name if downloaded directly (but I took them from another device too) and I presume it's something simple but I just can't figure it out.

**Comment by vshcherb** [comment, marked as answer]
You can just restart the app and it should work.

**Reply by vb0** [reply]
There's something weird going on, I downloaded for test directly on device Albania, being small (and coming soon in the alphabet and easy to type :-) ). OsmAnd updates properly if I take it out but when I put it back it gives no sign that it's there, both in Download Maps/LOCAL/Refresh and says *Download detailed Albania* when zooming in.  

I did kill and start it again, even restarted the phone.  

However, there is some awareness that the file is there ... if I try to "open it" from the file manager (I just noticed it's actually possible) it says *Map import error: A file with that name already exists*.  

Actually this thing that I've just noticed that OsmAnd registers probably these file types so you can "open" them is completely (and MUCH more elegantly) solving my problem but I realized now it's the same issue I've noticed just before starting this when trying some backups, even copying the whole net.osmand.plus directory (as described in https://osmand.net/help/faq.html ) and then bringing it back somehow isn't showing the maps for a fresh install (even following to the letter the whole procedure, setting the path to the new directory, then killing OsmAnd and bringing it back).  

I am keenly aware about all the shenanigans with storage access, this isn't it, OsmAnd can still read and write perfectly well from these directories, just won't see the files that were copied, as in not updating some index.

**Reply by tommycrock** [reply]
This is the shenanigans with storage access. It doesn't just apply to the folders but the files as well.  Osmand has to have written them in the download folder to be allowed to read them. So you need to import the files one by one (and that fails if the file is already present).

**Reply by vb0** [reply]
This is it indeed, thank you so much for the answer. It's unbelievable how Google is inventing all ways to make things not work, both blatantly and completely and partially and subtly. Any time now it'll become a feature if your device is old enough and/or the manufacturer is lazy enough not to push you over Android 10.
